backend/controllers/taskController.js

```diff
@@ -1,184 +1,167 @@
-const {isValidObjectId} = require("mongoose");
-const ObjectId = require("mongoose").Types.ObjectId;
 
 const createTask = asyncHandler(async (req, res) => {
-    const task = await new Task(req.body);
-    const createdTask = await task.save();
-    res.status(201).json(createdTask);
-
-    if (!createdTask) {
-        res.status(400);
-        throw new Error("Invalid Task Input");
-    }
+  const task = await new Task(req.body);
+  const createdTask = await task.save();
+  res.status(201).json(createdTask);
+
+  if (!createdTask) {
+    res.status(400);
+    throw new Error("Invalid Task Input");
+  }
 });
 
 const getTasks = asyncHandler(async (req, res) => {
-    const userId = req.id;
-    await Task.find({user_id: userId})
-        .then((docs) => {
-            res.status(200).json(docs);
-        })
-        .catch((err) => {
-            res.status(500);
-            throw new Error(`Could not fetch doc ${err}`);
-        });
-});
-
-const getTasksById = asyncHandler(async (req, res) => {
-    const taskId = req.params.id;
-    const userId = req.id;
-
-    const tasks = await Task.find({
-        _id: taskId,
-        user_id: userId,
-    })
-        .then((docs) => {
-            res.status(200).json(docs);
-        })
-        .catch((err) => {
-            res.status(500);
-            throw new Error(`Could not fetch doc ${err}`);
-        });
-    res.status(200).json(tasks);
-});
-
-const getTasksByDay = asyncHandler(async (req, res) => {
-    const date = req.params.day;
-    const userId = req.id;
-
-    let startDate = 0;
-
-    if (date.length === 8) {
-        // parse
-        const year = parseInt(date.substring(0, 4));
-        const month = parseInt(date.substring(4, 6));
-        const day = parseInt(date.substring(6, 8));
-
-        try {
-            startDate = new Date(year, month - 1, day);
-            endDate = new Date(year, month - 1, day + 2);
-        } catch (error) {
-            throw new Error("Invalid Date Input");
-        }
+  const startDateParam = req.query.start;
+  const endDateParam = req.query.end;
+  const userId = req.id;
+
+  let startDate = 0;
+  let endDate = 0;
+
+  if (startDateParam && endDateParam) {
+    if (startDateParam.length === 8 && endDateParam.length === 8) {
+      // parse
+      const year = parseInt(startDateParam.substring(0, 4));
+      const month = parseInt(startDateParam.substring(4, 6));
+      const day = parseInt(startDateParam.substring(6, 8));
+
+      const year_end = parseInt(endDateParam.substring(0, 4));
+      const month_end = parseInt(endDateParam.substring(4, 6));
+      const day_end = parseInt(endDateParam.substring(6, 8));
+
+      try {
+        startDate = new Date(year, month - 1, day);
+        endDate = new Date(year_end, month_end - 1, day_end + 1);
+      } catch (error) {
+        throw new Error("Invalid Date Input");
+      }
     }
-
-    // Check if its a valid date object
     const tasks = await Task.find({
```

Comment:
I would also place this in the `if statement` portion and then add an else to indicate that there's an Invalid Date Input (As of right now, I can theoretically write something like `http://localhost:5000/api/tasks?start=202203&end=123981239812398712398712397813102938102938012983` which then it should return Invalid Date Input

Reply:
Fixed: Changes are reflected in recent commits